pkg/kubelet/images/image_gc_manager.go

```diff
@@ -385,21 +385,28 @@ func (im *realImageGCManager) GarbageCollect(ctx context.Context, beganGC time.T
 	}
 
 	// If over the max threshold, free enough to place us at the lower threshold.
-	usagePercent := 100 - int(available*100/capacity)
-	if usagePercent >= im.policy.HighThresholdPercent {
+	usagePercent := 100 - float64(available)*100/float64(capacity)
```

Comment:
This seems suspect to me.

Why did you make this change?

Reply:
It doesn't really make sense to convert available or capacity to float64 when they are integers.

Was there an issue with the type casting?

Reply:
> This means any fractional portion is truncated, so a value like 69.9 becomes 69.

I think both are not right but am still curious why make the change here.

Reply:
If it helps, the previous int() cast didn't serve to truncate the fractional portion - it was already truncated by the int64 division - but to truncate to `int` before comparing to `im.policy.HighThresholdPercent`.

I guess that you're trying to convince yourself that this change to the message will not change the triggering logic of the GC through some weird truncation behavior. For example if the disk is >9PB, the numbers will be slightly truncated on conversion to float64 and the result could differ. But I recognize that this handwaving is not so compelling, so I'll restore the original integer division and reduce the precision of the error message.

Reply:
Thanks!

Yea I just don't want a subtle bug to pop up.